cmd/xl-v1-healing.go

```diff
@@ -639,12 +639,50 @@ func defaultHealResult(storageDisks []StorageAPI, errs []error, bucket, object s
 	return result
 }
 
+// Object is considered dangling/corrupted if any only if
+// total disks - total number of xl.json's are corrupted is lesser than number of data blocks.
+// total disks - total number of xl.json's are missing is lesser than number of data blocks.
+func (xl xlObjects) isObjectDangling(metaArr []xlMetaV1, errs []error) (validMeta xlMetaV1, ok bool) {
+	// We can consider an object data not reliable
+	// when xl.json is not found in read quorum disks.
+	var notFoundXLJSON, corruptedXLJSON int
+	for _, readErr := range errs {
+		if readErr == errFileNotFound {
+			notFoundXLJSON++
+		}
+	}
+
+	// We can consider an object data not reliable
+	// when xl.json is corrupted in read quorum disks.
+	for _, readErr := range errs {
```

Comment:
This loop can be eliminated by reusing the one above

Reply:
Done